qiskit/transpiler/passes/optimization/elide_swaps.py

```diff
@@ -0,0 +1,61 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""Remove the swaps followed by measurement (and adapt the measurement)."""
+
+from qiskit.circuit.library.standard_gates import SwapGate
+from qiskit.transpiler.basepasses import TransformationPass
+
+
+class ElideSwaps(TransformationPass):
+    r"""Remove :class:`~SwapGate`\s from a pre-layout circuit
+
+    This pass is intended to be run before a layout (mapping logical qubits
```

Comment:
```suggestion
    This pass is intended to be run before a layout (mapping virtual qubits
```

You used "virtual" further down, and we should probably make an effort to try and move towards "virtual" everywhere to avoid possible confusion with the QEC use of "logical qubit".

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/9523/commits/8f699e86e1c3aba1dc3102f6c6aeb4433ca05777